src/NHibernate/Linq/LinqExtensionMethods.cs

```diff
@@ -142,7 +53,34 @@ public static Task<bool> AnyAsync<TSource>(this IQueryable<TSource> source, Expr
 				throw new ArgumentNullException(nameof(predicate));
 			}
 			var provider = (INhQueryProvider)source.Provider;
-			var methodInfo = cachableQueryableMethods["AnyParam"].MakeGenericMethod(typeof(TSource));
+			var methodInfo = ReflectionCache.QueryableMethods.AnyWithPredicateDefinition.MakeGenericMethod(typeof(TSource));
+			var expression = new[] { source.Expression, Expression.Quote(predicate) };
+			return provider.ExecuteAsync<bool>(Expression.Call(null, methodInfo, expression), cancellationToken);
+		}
+
+		#endregion
+
+		#region AllAsync
```

Comment:
Was missing (recently added through [NH-3951](https://nhibernate.jira.com/browse/NH-3951)). Tests needing to be regenerated I guess.